### PR TITLE
Proto MVP

### PR DESCRIPTION
**Pull request by silviu-toderita**
-All Proto sensors working:
  -ECU
  -GPS
  -Thermocouple (not currently publishing as waiting on Luke to let us know how many we will be using for each vehicle)
-Added comments and documentation throughout code for better documentation
-Added some constants to main file for ease of modification
-Added custom makefile to both proto and urban so that our projects will compile with custom library file structure

### CODE REVIEW

File: proto/src/main.cpp [hunk 1]
```diff
@@ -1,73 +1,89 @@
...
-    // Blink the LED
-    if (led_state & (millis() - last_blink >= BLINK_INTERVAL_ON)){
-        led_state = LOW;
-        last_blink = millis();
-    }else if(!led_state & (millis() - last_blink >= BLINK_INTERVAL_OFF)){
-        led_state = HIGH;
-        last_blink = millis();
+    ecu.handle();
+    gps.handle();
+    thermoA.handle();
```
Comment (lcfyi) [inline proto/src/main.cpp:79]: Would you be able to time (just diff the `millis()` before and after each call) each of these handlers? Ideally they should be quite fast (so it's not a problem), but if there's a tangible delay between each then our single datapoint for all the data from this timestamp would not line up.

File: proto/src/main.cpp [hunk 2]
```diff
@@ -1,73 +1,89 @@
...
-JsonMaker json_maker;
-Sensor_RSSI rssi;
-Sensor_ECU ecu(&Serial1);
-
-
-uint32_t last_blink = 0; // Time of last blink
-uint32_t last_publish = 0; // Time of last publish
-boolean led_state = LOW;
+// Disable publish when not necessary during development to reduce data usage
+#define PUBLISH_ENABLED             1
```
Comment (lcfyi) [inline proto/src/main.cpp:11]: I like this a lot. One suggestion would be to also wrap the `Serial.print` statements into a macro that's enabled/disabled based on this flag. For example, something like [this](https://stackoverflow.com/questions/31860255/console-output-only-in-debug-mode-c).

**Review by lcfyi — COMMENTED** [review]
Good stuff! Left two small comments but nothing huge.

**Comment by lcfyi** [comment]
I also just noticed these PRs were targetting the `master` branch rather than `develop`. I'll bring them up to parity after this is merged.

You'll probably also want to rebase this branch off of master, which will introduce a couple merge conflicts. Let me know if you need a hand dealing with those.

File: proto/src/main.cpp (hunk 1 above)
Comment (silviu-toderita) [reply proto/src/main.cpp]: I added optional timing that you can enable with a flag in the latest commit. Here are the values we're getting (in MICROseconds, not miliseconds): 

TIME ELAPSED FOR: 
ECU Polling:                   537us
GPS Polling:                11158us
Thermo Polling:             673us
Build JSON Message: 1559us

File: proto/src/main.cpp (hunk 2 above)
Comment (silviu-toderita) [reply proto/src/main.cpp:11]: Done, though added a new flag for this as we will want to output debug messages over serial AND have publish enabled during testing also

**Pull request merged**